napari/layers/shapes/shapes.py

```diff
@@ -2160,13 +2148,22 @@ def _add_shapes(
             same length as the length of `data` and each element will be
             applied to each shape otherwise the same value will be used for all
             shapes.
-        z_refresh : bool
-            If set to true, the mesh elements are reindexed with the new z order.
-            When shape_index is provided, z_refresh will be overwritten to false,
-            as the z indices will not change.
-            When adding a batch of shapes, set to false  and then call
-            ShapesList._update_z_order() once at the end.
         """
+        if n_new_shapes > 0:
```

Comment:
It is None by default. A comparison of None with 0 will end with an error. Did you want to use 0 as the default value? Or maybe move to begin and not provide a default value?

Reply:
I changed the default to 0, consistent with `number_of_shapes`.